pkg/apis/options/providers.go

```diff
@@ -92,6 +92,9 @@ type Provider struct {
 	// The code challenge method
 	CodeChallengeMethod string `json:"code_challenge_method,omitempty"`
 
+	// Additional claims to be obtained from the `id_token`.
+	AdditionalClaims []string `json:"additionalClaims,omitempty"`
+
```

Comment:
As this implementation uses the `ClaimExtractor`, this is not restricted to the `id_token`. It actually would extract claims from the `id_token` or the `userinfo` endpoint of the upstream IDP if the `profileURL` is configured.